pkg/ruler/manager.go

```diff
@@ -131,10 +131,10 @@ func (r *DefaultMultiTenantManager) SyncRuleGroups(ctx context.Context, ruleGrou
 // syncRulesToManager maps the rule files to disk, detects any changes and will create/update
 // the user's Prometheus Rules Manager. Since this method writes to disk it is not safe to call
 // concurrently for the same user.
-func (r *DefaultMultiTenantManager) syncRulesToManager(ctx context.Context, user string, groups rulespb.RuleGroupList) {
+func (r *DefaultMultiTenantManager) syncRulesToManager(ctx context.Context, user string, groups rulespb.RuleGroupList, limit int) {
```

Comment:
suggestion: instead of passing an int through a few functions is it possible to keep a `RulerLimits` instance in the `DefaultMultiTenantManager`? I don't think the value that implements `RulerLimits` changes throughout the lifecycle of the ruler, so it should be safe to set it once on creation and reuse it.

Reply:
Thanks for the suggestion @dimitarvdimitrov, implemented it.